src/qibocal/protocols/two_qubit_state_tomography.py

```diff
@@ -0,0 +1,300 @@
+            basis_circuit,
+            qubits,
+            backend,
+            nshots=params.nshots,
+            transpiler=transpiler,
+        )
+
+        for i, pair in enumerate(targets):
+            frequencies = results.frequencies(registers=True)[f"reg{i}"]
+            simulation_probabilities = simulation_result.probabilities(
+                qubits=(2 * i, 2 * i + 1)
+            )
+            data.register_qubit(
+                TomographyType,
+                pair + (basis1, basis2),
+                {
+                    "frequencies": np.array([frequencies[i] for i in OUTCOMES]),
+                    "simulation_probabilities": simulation_probabilities,
+                },
+            )
+            if basis1 == "Z" and basis2 == "Z":
+                nqubits = basis_circuit.nqubits
+                traced_qubits = tuple(
+                    q for q in range(nqubits) if q not in (2 * i, 2 * i + 1)
+                )
+                data.ideal[pair] = simulator.partial_trace(
+                    simulation_result.state(), traced_qubits, nqubits
+                )
+
+    return data
+
+
+def rotation_matrix(basis):
+    """Matrix of the gate implementing the rotation to the given basis.
+
+    Args:
+        basis (str): One of Pauli basis: X, Y or Z.
+    """
+    backend = NumpyBackend()
+    if basis == "Z":
+        return np.eye(2, dtype=complex)
+    return getattr(gates, basis)(0).basis_rotation().matrix(backend)
+
+
+def project_psd(matrix):
+    """Project matrix to the space of positive semidefinite matrices."""
+    s, v = np.linalg.eigh(matrix)
+    s = s * (s > 0)
+    return v.dot(np.diag(s)).dot(v.conj().T)
+
+
+def _fit(data: StateTomographyData) -> StateTomographyResults:
+    """Post-processing for State tomography."""
+    rotations = [
+        np.kron(rotation_matrix(basis1), rotation_matrix(basis2))
+        for basis1, basis2 in TWO_QUBIT_BASIS
+    ]
+
+    # construct the linear transformation from density matrix to Born-probabilities
+    measurement = np.zeros((0, 16))
+    for rotation in rotations:
+        channel = np.kron(rotation, np.conj(rotation))
+        measure_channel = channel[np.eye(4, dtype=bool).flatten(), :]
+        measurement = np.concatenate((measurement, measure_channel))
+
+    # invert to get linear transformation from Born-probabilities to density matrix
+    inverse_measurement = np.linalg.pinv(measurement)
+
+    # calculate Born-probabilities vector from measurements (frequencies)
+    probabilities = defaultdict(lambda: np.empty(measurement.shape[0]))
+    for (qubit1, qubit2, basis1, basis2), value in data.data.items():
+        frequencies = value["frequencies"]
+        ib = TWO_QUBIT_BASIS.index((basis1, basis2))
+        probabilities[(qubit1, qubit2)][4 * ib : 4 * (ib + 1)] = frequencies / np.sum(
+            frequencies
+        )
+
+    # reconstruction
+    backend = NumpyBackend()
+    results = StateTomographyResults({}, {}, {}, {}, {})
```

Comment:
```suggestion
    results = StateTomographyResults()
```
If you initialize each attribute of `StateTomograpyResults` as an empty dict you avoid to do this. I believe that it would improve readability.

Reply:
288c8208cdc0afa90867bc867eb94210675a5cf0